### PR TITLE
Add data normalization in QP and clean QP, LS and EGO

### PR DESCRIPTION
**Pull request by Paul-Saves**
Add data normalization in QP
xlimits is no longer a parameter of QP and LS
self.work_in_folded_space is now generalized and properly checked

### CODE REVIEW

**Comment by Paul-Saves** [comment]
Thanks to @relf, following issues of #394

File: smt/applications/mixed_integer.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def __init__(
         self._cat_kernel_comps = cat_kernel_comps
         self._xtypes = xtypes
         self._xlimits = xlimits
-        self._surrogate.options["xlimits"] = self._xlimits
+        if "xlimits" in self._surrogate.options:
+            self._surrogate.options["xlimits"] = self._xlimits
```
Comment (relf) [inline smt/applications/mixed_integer.py:108]: I am still puzzled by this `xlimits` option. Why not use the values themselves to normalize when using Gower distance? Like `standardization`?

File: smt/surrogate_models/qp.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import scipy
 from smt.surrogate_models.surrogate_model import SurrogateModel
 from smt.utils.caching import cached_operation
-
+from smt.utils.kriging_utils import standardization
```
Comment (relf) [inline smt/surrogate_models/qp.py:14]: Well, `standardization` should be moved in a another module in utils? `utils/misc.py` ?

File: smt/surrogate_models/qp.py [hunk 3]
```diff
@@ -118,7 +118,7 @@ def _predict_derivatives(self, x, kx):
             Derivative values.
         """
         dim = self.nx
-
+        x = (x -self.X_offset)/self.X_scale
```
Comment (relf) [inline smt/surrogate_models/qp.py:121]: Seems you need to pass `black`.

**Review by relf — CHANGES_REQUESTED** [review]

**Comment by Paul-Saves** [comment]
Ok, j'ai passé `black`.  Le reste sera traité dans la prochaine PR

**Pull request merged**